libpldmresponder/fru.cpp

```diff
@@ -184,23 +200,24 @@ void FruImpl::buildFRUTable()
                     pldm_entity entity{};
                     if (objToEntityNode.contains(object.first.str))
                     {
-                        pldm_entity_node* node =
-                            objToEntityNode.at(object.first.str);
-
+                        pldm_entity_node* node = nullptr;
+                        pldm_find_entity_ref_in_tree(
+                            entityTree, objToEntityNode.at(object.first.str),
+                            &node);
                         entity = pldm_entity_extract(node);
                     }
 
                     auto recordInfos = parser.getRecordInfo(interface.first);
-                    populateRecords(interfaces, recordInfos, entity);
-
+                    populateRecords(interfaces, recordInfos, entity,
+                                    object.first);
                     associatedEntityMap.emplace(object.first, entity);
                     break;
                 }
                 catch (const std::exception& e)
                 {
                     error(
-                        "Config JSONs missing for the item '{INTERFACE}', error - {ERROR}",
-                        "INTERFACE", interface.first, "ERROR", e);
+                        "Config JSONs missing for the item interface type, interface = {INTF}",
+                        "INTF  and ERROR={ERR_EXCEP}", interface.first, "ERR_EXCEP", e);
```

Comment:
redundant INTF  tag,

Reply:
this seems like a bad merge, we need the upstream change only here.